src/app/shared/product.service.ts

```diff
@@ -0,0 +1,23 @@
+import { Injectable } from '@angular/core';
+import { HttpClient } from '@angular/common/http';
+
+@Injectable({
+  providedIn: 'root'
+})
+export class ProductService {
+
+  constructor( private http: HttpClient) { }
+
+  addProduct(product) {
+    console.log(product)
+    return this.http.post('http://localhost:3003/api/products', product)
```

Comment:
Please add back end base url `http://localhost:3003/api` to environments and use this here from environments